pkg/antctl/raw/migrate/convertnetworkpolicy/command.go

```diff
@@ -0,0 +1,488 @@
+	return nil
+}
+
+func (c *converter) convertGlobalNetworkPolicy() error {
+	gnps, err := c.calicoClient.ProjectcalicoV3().GlobalNetworkPolicies().List(context.Background(), metav1.ListOptions{})
+	if err != nil {
+		klog.Errorf("Failed to list GlobalNetworkPolicies: %v", err)
+		return err
+	}
+	for _, gnp := range gnps.Items {
+		klog.Infof("Converting GlobalNetworkPolicy %s", gnp.Name)
+		acnp := v1beta1.ClusterNetworkPolicy{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: gnp.Name,
+				Annotations: map[string]string{
+					convertFromCalicoAnnotation: "true",
+				},
+			},
+		}
+
+		if acnp.Spec.AppliedTo, err = convertSelectorToAntreaAppliedTo(gnp.Spec.Selector, gnp.Spec.NamespaceSelector); err != nil {
+			return err
+		}
+		if acnp.Spec.Ingress, err = convertCalicoRuleToAntreaRule(gnp.Spec.Ingress); err != nil {
+			return err
+		}
+		if acnp.Spec.Egress, err = convertCalicoRuleToAntreaRule(gnp.Spec.Egress); err != nil {
+			return err
+		}
+		if gnp.Spec.Order == nil {
+			acnp.Spec.Priority = defaultPriority
+		} else {
+			acnp.Spec.Priority = *gnp.Spec.Order
+		}
+
+		if _, err := c.antreaClient.CrdV1beta1().ClusterNetworkPolicies().Create(context.Background(), &acnp, metav1.CreateOptions{}); err != nil {
+			klog.Errorf("Failed to create ClusterNetworkPolicy %s: %v", acnp.Name, err)
+			return err
+		}
+	}
+	return nil
+}
+
+func convertCalicoRuleToAntreaRule(rule []v3.Rule) ([]v1beta1.Rule, error) {
+	var antreaRules []v1beta1.Rule
+
+	for _, r := range rule {
+		antreaRule := v1beta1.Rule{}
+		switch r.Action {
+		case v3.Allow:
+			antreaRule.Action = migrate.ToPtr(v1beta1.RuleActionAllow)
+		case v3.Log:
+			antreaRule.EnableLogging = true
+			antreaRule.Action = migrate.ToPtr(v1beta1.RuleActionAllow)
+		case v3.Deny:
+			antreaRule.Action = migrate.ToPtr(v1beta1.RuleActionReject)
+		case v3.Pass:
+			antreaRule.Action = migrate.ToPtr(v1beta1.RuleActionPass)
+		}
+		var err error
+		if antreaRule.From, err = convertEntityRuleToAntreaNetworkPolicyPeer(r.Source); err != nil {
+			return nil, err
+		}
+		if antreaRule.To, err = convertEntityRuleToAntreaNetworkPolicyPeer(r.Destination); err != nil {
+			return nil, err
+		}
+		if antreaRule.Ports, err = convertCalicoPortToAntreaPort(r); err != nil {
+			return nil, err
+		}
+		antreaRule.ToServices = convertCalicoServiceToAntreaService(r)
+		antreaRules = append(antreaRules, antreaRule)
+	}
+
+	if len(rule) != 0 {
+		antreaRules = append(antreaRules,
+			v1beta1.Rule{Action: migrate.ToPtr(v1beta1.RuleActionReject)})
+	}
+	return antreaRules, nil
+}
+
+func convertCalicoPortToAntreaPort(rule v3.Rule) ([]v1beta1.NetworkPolicyPort, error) {
+	antreaPorts := []v1beta1.NetworkPolicyPort{}
+	for _, srcPort := range rule.Source.Ports {
+		antreaPort := v1beta1.NetworkPolicyPort{
+			SourcePort:    migrate.ToPtr(int32(srcPort.MinPort)),
+			SourceEndPort: migrate.ToPtr(int32(srcPort.MaxPort)),
+		}
+		if rule.Protocol != nil {
+			antreaPort.Protocol = (*v1.Protocol)(&rule.Protocol.StrVal)
```

Comment:
Seems Calico has more protocols than Antrea, you should check it before conversion.

Reply:
A check is added in `precheck` subcommand.